admin_manual/enterprise/external_storage/s3_swift_as_primary_object_store_configuration.rst

```diff
@@ -1,159 +1,167 @@
-=============================================================
-Configuring S3 and OpenStack Swift Objects as Primary Storage
-=============================================================
+=================================
+Configuring S3 as Primary Storage
+=================================
 
-In ownCloud Enterprise edition, you can configure S3 objects as primary 
-storage. This replaces the default ownCloud ``owncloud/data`` directory. You may 
-still need to keep the ``owncloud/data`` directory for these reasons:
+In ownCloud Enterprise edition administrators can configure Amazon S3 objects as the primary storage.
+Doing this replaces the default ownCloud ``owncloud/data`` directory.
+
+You may still need to keep the ``owncloud/data`` directory for these reasons:
 
 * The ownCloud log file is saved in the data directory
 * Legacy apps may not support using anything but the ``owncloud/data`` directory
 
-You can move your logfile by changing its location in ``config.php``. You may still need 
-``owncloud/data`` for backwards compatibility with some apps.
+Administrators can move their ownCloud log file, by changing its location in ``config.php``.
+However, even when this is done, ``owncloud/data`` may still be required for backward compatibility with some apps.
+
+
+`The S3 Object Storage app`_ (``files_primary_s3``) is the preferred way to provide S3 storage support,
+However, `the Object Storage Support app`_ (``objectstore``) is still available.
+
+.. note:: OpenStack Swift has been deprecated.
+
+When using ``files_primary_s3``, the Amazon S3 bucket needs to be created manually according to the developer documentation, and versioning needs to be enabled.
+
+.. note:: ownCloud GmbH provides consulting for migrations from ``objectstore`` to ``files_primary_s3``.
+
+.. note:: The source of the S3 Object Storage app `is available on gitHub`_.
 
 Implications
 ------------
-This configuration needs to be applied before the first login of any user – including the 
-admin user. Otherwise, you will run into an error condition, because ownCloud cannot find 
-the user's files anymore.
 
-ownCloud in object store mode expects exclusive access to the object store 
-container, because it only stores the binary data for each file. The metadata 
-are kept in the local database for performance reasons.
+- Apply this configuration before the first login of any user – including the admin user; otherwise, ownCloud can no longer find the user's files.
```

Comment:
This is a bit tricky.
If I am setting up a new instance, I create the admin user and logon with that user for the first time. Then I do the preparing steps for the environment including customisation of config.php. That text above conflicts with this situation - or did I miss something?
Something differnt when migrating an existing data folder to object store as there are existing users to be migrated. What is the situation / plan there?

Reply:
@DeepDiver1975?

Reply:
It has Always been this was. No problem in any instance